kyo-core/shared/src/main/scala/kyo/StreamCoreExtensions.scala

```diff
@@ -61,6 +61,33 @@ object StreamCoreExtensions:
                             _ <- emitMaybeChunksFromChannel(channel)
                         yield ()
 
+        /** Creates a stream from an iterator.
+          *
+          * @param v
+          *   Iterator to create a stream from
+          * @param bufferSize
+          *   Size of the buffer that the iterator will write to and the stream will read from
+          */
+        def fromIterator[V, S](v: => Iterator[V] < S, bufferSize: Int = defaultAsyncStreamBufferSize)(using
```

Comment:
We typically avoid allowing effects in value parameters like the iterator here. It reduces complexity, overhead, and makes reasoning about where effects happen easier. Could you take the `Iterator` directly?